### PR TITLE
ZCS-4583 Implement GetInfoRequest in the AccountInfoRepository class

### PR DESCRIPTION
**Pull request by chinmay15**
**Problem:** implement GetInfoRequest in gql

**Fix:** Implemented GetInfoRequest in gql as "infoGet"

**Testing done:**
- Tested infoGet from GraphiQL manually with some of the options manually.

**Testing needs to be done by QA:**
- Test infoGet completely in comparison with soap GetInfoRequest.

**Linked PR:**
https://github.com/Zimbra/zm-gql/pull/34

### CODE REVIEW

File: common/src/java/com/zimbra/common/gql/GqlConstants.java [hunk 1]
```diff
@@ -303,4 +303,103 @@
     public static final String LAST_ACCESSED = "lastAccessed";
     public static final String USER_AGENT = "userAgent";
     public static final String REQUEST_IP_ADDRESS = "requestIPAddress";
+
+    // get info constants
+    public static final String GET_INFO_REQUEST = "GetInfoRequest";
```
Comment (desouzas) [inline common/src/java/com/zimbra/common/gql/GqlConstants.java]: Prefix all of the class names you added here with `CLASS_`. for consistency with the rest. e.g. `CLASS_GET_INFO_REQUEST`, or `CLASS_ACCOUNT_ZIMLET_INFO`, etc (15~ occurrences)

File: soap/src/java/com/zimbra/soap/account/type/AccountZimletConfigInfo.java [hunk 2]
```diff
@@ -109,20 +114,28 @@ public AccountZimletConfigInfo() {
     public void setLabel(String label) { this.label = label; }
     public void setGlobal(AccountZimletGlobalConfigInfo global) { this.global = global; }
     public void setHost(AccountZimletHostConfigInfo host) { this.host = host; }
+    @GraphQLQuery(name=GqlConstants.NAME, description="Zimlet name")
```
Comment (desouzas) [inline soap/src/java/com/zimbra/soap/account/type/AccountZimletConfigInfo.java]: Put all the gql annotations below other annotations in this file.

File: soap/src/java/com/zimbra/soap/account/type/AccountZimletContext.java [hunk 3]
```diff
@@ -79,10 +85,15 @@ public static AccountZimletContext createForBaseUrlPriorityAndPresence(
     @Override
     public void setZimletPresence(String zimletPresence) { this.zimletPresence = zimletPresence; }
 
+    @GraphQLQuery(name=GqlConstants.ZIMLET_BASE_URL, description="Zimlet Base URL")
+    @GraphQLNonNull
```
Comment (desouzas) [inline soap/src/java/com/zimbra/soap/account/type/AccountZimletContext.java]: Put all the gql annotations below other annotations in this file.

File: soap/src/java/com/zimbra/soap/account/type/AccountZimletProperty.java [hunk 4]
```diff
@@ -65,8 +70,10 @@ public static AccountZimletProperty createForNameAndValue(String name, String va
     public void setName(String name) { this.name = name; }
     @Override
     public void setValue(String value) { this.value = value; }
+    @GraphQLQuery(name=GqlConstants.NAME, description="Name")
```
Comment (desouzas) [inline soap/src/java/com/zimbra/soap/account/type/AccountZimletProperty.java]: Put all the gql annotations below other annotations in this file.

File: soap/src/java/com/zimbra/soap/account/type/AccountZimletHostConfigInfo.java [hunk 5]
```diff
@@ -78,6 +83,7 @@ public void addProperty(AccountZimletProperty property) {
         this.properties.add(property);
     }
 
+    @GraphQLQuery(name=GqlConstants.NAME, description="Name")
```
Comment (desouzas) [inline soap/src/java/com/zimbra/soap/account/type/AccountZimletHostConfigInfo.java]: Put all the gql annotations below other annotations in this file.

File: soap/src/java/com/zimbra/soap/account/type/AccountZimletDesc.java [hunk 6]
```diff
@@ -140,18 +145,25 @@ public void addElement(Object element) {
         this.elements.add(element);
     }
 
+    @GraphQLQuery(name=GqlConstants.NAME, description="Zimlet name")
```
Comment (desouzas) [inline soap/src/java/com/zimbra/soap/account/type/AccountZimletDesc.java]: Put all the gql annotations below other annotations in this file.

File: soap/src/java/com/zimbra/soap/account/message/GetInfoRequest.java [hunk 7]
```diff
@@ -37,6 +41,7 @@
  *     sections you want in the "sections" attr.
  */
 @XmlRootElement(name=AccountConstants.E_GET_INFO_REQUEST)
+@GraphQLType(name=GqlConstants.CLASS_GET_INFO_REQUEST, description="Get information about an account")
```
Comment (shrikant-prasad) [inline soap/src/java/com/zimbra/soap/account/message/GetInfoRequest.java]: don't need GQL annotations in this class as it's not used in the graphql schema.

**Comment by desouzas** [comment]
> 
> 
> I don't think there is any importance of the annotations ordering here.

Possibly true - I just remember having an issue once where the annotation was ignored when placed above another one. Possibly I misunderstood.

**Comment by chinmay15** [comment]
merged in develop

commit 5bf90611c5461c971edfc475cfbdbe1782fb1b06 (HEAD -> develop, origin/develop, origin/HEAD)
Author: Chinmay Pathak <15.chinmay@gmail.com>
Date:   Mon Oct 8 16:34:52 2018 +0530

    ZCS-4583 Implement GetInfoRequest in the AccountInfoRepository class
    
    Corrected validations
    
    Resolving review comments

**Pull request merged**